pkg/registry/debian/debian.go

```
@@ -0,0 +1,156 @@
+type HTTPRegistry struct {
+	Client httpx.BasicClient
+}
+
+func (r HTTPRegistry) get(ctx context.Context, url string) (io.ReadCloser, error) {
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	resp, err := r.Client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	if resp.StatusCode != 200 {
+		return nil, errors.Errorf("fetching artifact: %v", resp.Status)
+	}
+	return resp.Body, nil
+}
+
+func PoolURL(component, name, artifact string) string {
+	prefixDir := name[0:1]
+	// Most packages are in a prefix dir matching their first letter. The exception are lib* files which are in dirs matching lib*.
+	if strings.HasPrefix(name, "lib") {
+		prefixDir = name[0:4]
+	}
+	return registryURL + fmt.Sprintf("/pool/%s/%s/%s/%s", component, prefixDir, name, artifact)
+}
+
+func guessDSCURL(component, name, version string) string {
+	cleanVersion := binaryReleaseRegexp.ReplaceAllString(version, "")
+	return PoolURL(component, name, fmt.Sprintf("%s_%s.dsc", name, cleanVersion))
+}
+
+func parseDSC(r io.ReadCloser) (*DSC, error) {
+	b := bufio.NewScanner(r)
+	if !b.Scan() {
+		return nil, errors.New("failed to scan .dsc file")
+	}
+	// Skip PGP signature header.
+	if strings.HasPrefix(b.Text(), "-----BEGIN PGP SIGNED MESSAGE-----") {
+		b.Scan()
+	}
+	d := DSC{}
+	stanza := ControlStanza{Fields: map[string][]string{}}
+	var lastField string
+	for {
+		// Check for PGP signature footer.
+		if strings.HasPrefix(b.Text(), "-----BEGIN PGP SIGNATURE-----") {
+			break
+		}
+		line := b.Text()
+		if strings.TrimSpace(line) == "" {
+			// Handle empty lines as stanza separators.
+			if len(stanza.Fields) > 0 {
+				d.Stanzas = append(d.Stanzas, stanza)
+				stanza = ControlStanza{Fields: map[string][]string{}}
+			}
+		} else if strings.HasPrefix(line, " ") || strings.HasPrefix(line, "\t") {
+			// Handle continuation lines.
+			if lastField != "" {
+				stanza.Fields[lastField] = append(stanza.Fields[lastField], strings.TrimSpace(line))
+			}
```

i think the else case should return an error